src/Logic/ImplicationGraph/Simplify.hs

```diff
@@ -0,0 +1,50 @@
+module Logic.ImplicationGraph.Simplify where
+
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import           Data.Maybe
+import qualified Logic.Var as V
+import           Logic.Formula
+import           Logic.ImplicationGraph
+
+
+cartesianProduct :: (a -> b -> c) -> [a] -> [b] -> [c]
+cartesianProduct f as bs = [ f a b | a <- as, b <- bs ]
+
+
+conjunction :: Edge -> Edge -> Edge
+conjunction edge edge' =
+  let
+    onlyFreeVars = \case
+      V.Bound _ _ -> Nothing
+      var @ (V.Free path time _) -> Just (var, path, time)
+
+    furthestVarMap mapping (var, path, time) =
```

Comment:
This should probably just use `_edgeMap edge` instead of creating a map over all variables in this edge.